### DISCUSSION TITLE
Quick terminal detection in shell profiles

### DISCUSSION THREAD

**Discussion by choovick** [ghostty-org/ghostty#3985] [Feature Requests, Ideas] [answered]
Amazing launch, congrats to all developers!

**Use case:** when zsh session starts I have a specific terminal checks to start/attach my default TMUX session: https://github.com/choovick/.dotfiles/blob/main/stow/zsh/.zshrc#L291

But quick terminal treated as singleton, and I don't want a tmux session, or I might want another tmux profile, but I'm not able to detect it.

I see only the following env vars:

```shell
GHOSTTY_RESOURCES_DIR=/Applications/Ghostty.app/Contents/Resources/ghostty
GHOSTTY_SHELL_INTEGRATION_NO_SUDO=1
GHOSTTY_BIN_DIR=/Applications/Ghostty.app/Contents/MacOS
```

Is it currently possible to add env var on quick terminal launch? 
My keybind is `keybind = global:super+shift+alt+ctrl+f1=toggle_quick_terminal`

**Comment by integrii** [comment]
I would love this as well.  Anyone who has their `tmux` attach automatically would have an easier time getting the quick terminal behaving properly.

I made this work with the following until then because the terminal it created always had a height of 12 lines.

```sh
if [ -z "$TMUX" ]; then
  if [[ $(tput lines) != 12 ]]; then
    tmux attach || tmux new-session
  fi
fi
```

**Reply by choovick** [reply]
Thanks @integrii 

I tried it, it kinda works. the issue I think number of lines determined based on screen height/resolution. I have multiple monitors and depending where 1st instance in launched `tput lines` will differ. I guess I can check against a set for all my screens.

I also did not find any settings to set that desired lines hight: https://ghostty.org/docs/config/reference#quick-terminal-position

**Comment by cloud-on-prem** [comment]
This would be such a great addition. My personal use case for Quick Terminal is to open my Obsidian Notes in neovim for quick note-taking. It would be awesome if my shell is aware of the quick terminal status.

**Comment by 00-kat** [comment, marked as answer]
This should probably be considered a duplicate of #2353.

**Comment by birkhofflee** [comment]
How does this relate to scripting or an API? It's as simple as setting a environment variable (e.g. `GHOSTTY_IS_QUICK_TERMINAL=true`).

I don't write Swift or Zig. I've traced the code execution path on GitHub. It looks simple as providing the env in `environmentVariables`[1] in the `baseConfig` when initializing a new terminal for Quick Terminal[2].

[1] https://github.com/ghostty-org/ghostty/blob/9dc2e5978f6ee69f4d784b61865589b502c4012d/macos/Sources/Ghostty/SurfaceView.swift#L438
[2] https://github.com/ghostty-org/ghostty/blob/9dc2e5978f6ee69f4d784b61865589b502c4012d/macos/Sources/Features/QuickTerminal/QuickTerminalController.swift#L345

**Reply by rhodes-b** [reply]
with the scriping API you will be able to ask ghostty information about the current surface ie if its a quick terminal

**Reply by choovick** [reply]
> scriping API

Yea, reasonable approach, is that flag available there already? (never used it yet)

**Reply by choovick** [reply]
> > scriping API
> 
> Yea, reasonable approach, is that flag available there already? (never used it yet)

oh, looks like its a work in progress... no rush, I still have couple dozen years in my life :D

**Comment by viniciussoares** [comment]
I've implemented this feature in #9673.

Added `GHOSTTY_QUICK_TERMINAL=1` environment variable that gets set in quick terminal sessions (both macOS and Linux/GTK).

I know @rhodes-b mentioned this was planned for the scripting API, and I think that's still valuable for more advanced use cases! This environment variable approach just provides a simpler solution for the immediate shell configuration use case (like skipping tmux auto-attach). Both can definitely coexist. 🙂

Example usage:
```zsh
if [[ -z "$TMUX" ]] && [[ -z "$GHOSTTY_QUICK_TERMINAL" ]]; then
  tmux attach || tmux new
fi
```

Happy to adjust the implementation based on feedback!

**Reply by choovick** [reply]
Thank you! change looks pretty simple and hope there will be no resistance to merge :) I might start using quick terminal...
